src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,145 @@
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.ctre.phoenix.sensors.CANCoder;
+import com.ctre.phoenix.sensors.CANCoderConfiguration;
+import com.ctre.phoenix.sensors.SensorTimeBase;
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.SubsystemChecker;
+import frc.robot.SubsystemChecker.SubsystemType;
+import frc.robot.config.Config;
+
+public class ArmSubsystem extends SubsystemBase {
+
+  private static final MotorType motorType = MotorType.kBrushless;
+  private static final SparkMaxAbsoluteEncoder.Type encAbsType = SparkMaxAbsoluteEncoder.Type.kDutyCycle;
+  
+  public CANCoder m_absoluteTopArmEncoder;
+  public CANCoder m_absoluteBottomArmEncoder;
+
+
+  private static ArmSubsystem instance = null;
+
+
+  public final CANSparkMax m_topArm;
+  public final CANSparkMax m_bottomArm;
+  public SparkMaxPIDController m_pidControllerTopArm;
+  public SparkMaxPIDController m_pidControllerBottomArm;
+  public final double kP = 0.6;
+  public final double kI = 0;
+  public final double kD = 0; 
+  public final double kIz = 0;
+  public final double kFF = 0;
+  public final double kMaxOutput = 0.5;
+  public final double kMinOutput = -0.5;
+
+  /** Creates a new ArmSubsystem. */
+  public ArmSubsystem() {
+    m_topArm = new CANSparkMax(Config.CANID.TOP_ARM_SPARK_CAN_ID, motorType);
+    m_bottomArm = new CANSparkMax(Config.CANID.BOTTOM_ARM_SPARK_CAN_ID, motorType);
+    CANCoderConfiguration config = new CANCoderConfiguration();
+    m_topArm.restoreFactoryDefaults();
+    m_bottomArm.restoreFactoryDefaults();
+    // m_topArm.setInverted(true);
+    // m_bottomArm.setInverted(true);
+
+    // set units of the CANCoder to radians, with velocity being radians per second
+    config.unitString = "rad";
+    config.sensorTimeBase = SensorTimeBase.PerSecond;
+
+    m_absoluteTopArmEncoder = new CANCoder(16);
+    m_absoluteBottomArmEncoder = new CANCoder(17);
+
+    m_absoluteTopArmEncoder.configAllSettings(config);
+    m_absoluteBottomArmEncoder.configAllSettings(config);
+
+    m_pidControllerTopArm = m_topArm.getPIDController();
+    m_pidControllerBottomArm = m_bottomArm.getPIDController();
+
+    // setting PID coefficients for top arm
+    m_pidControllerTopArm.setP(kP);
+    m_pidControllerTopArm.setI(kI);
+    m_pidControllerTopArm.setD(kD);
+    m_pidControllerTopArm.setIZone(kIz);
+    m_pidControllerTopArm.setFF(kFF);
+    m_pidControllerTopArm.setOutputRange(kMinOutput, kMaxOutput);
+    m_pidControllerTopArm.setSmartMotionMaxAccel(Math.PI/6, 0); // maybe accel and velocity should be in rpm instead of radians/second ?
```

Comment:
The velocity and acceleration is in radians/second and radians/second^2 because of the conversation factor that is set. I highly encourage we stick with these units because they are a lot easier to understand than RPM and RPM per second.

I did just realize, we might need to also set the velocity conversion factor on top of the position conversation factor that we are already setting. The value for the velocity conversation factor is the same as the position conversion factor except you need to divide it by 60. (an example of this can be found in Config which is used for the swerve motors [https://github.com/FRC2706/2023-2706-Robot-Code/blob/ae44e6fb3c47a503f05bc270d8e8da30eed498a2/src/main/java/frc/robot/config/Config.java#L162](https://github.com/FRC2706/2023-2706-Robot-Code/blob/ae44e6fb3c47a503f05bc270d8e8da30eed498a2/src/main/java/frc/robot/config/Config.java#L162))

Reply:
I totally agree to use a more reasonable unit by the conversion factor. 
I am just wondering if the MaxAccel and MaxVelocity are affected by the conversion factor. (From their APIs document, it is not said that. )

Reply:
I just found some messages from a very well know person in WPILib that SmartMotion is apparently not written well and he recommended not using it.

We might need to stick with just position PID.